### PR TITLE
feat(rtc_auto_mode_manager): add rtc_auto_mode_manager and fix auto mode behavior

### PR DESCRIPTION
**Pull request by rej55**
Signed-off-by: Fumiya Watanabe <rej55.g@gmail.com>

## Description

`RTC Auto Approver` had a problem that behavior velocity planning module inserts stop line for a moment because `RTC Auto Approver` sends a command after receiving RTC status from the planning module.

To solve this problem, I add the function to approve automatically to `rtc_interface`.
Furthermore, I removed approving function from `RTC Auto Approver`.
So, `RTC Auto Approver` just has the function to set default value to enable auto mode.
To clarify the function, I rename `RTC Auto Approver` to `RTC Auto Mode Manager`.

The behavior of auto mode is changed as following:
- If it receives `cooperate_commands` when auto mode is enabled
  - It plans following received `cooperate_commands` and turns off the auto mode
  - This behavior is just applied to the target that has same UUID

## Tests performed

Tested in planning simulator.

https://user-images.githubusercontent.com/10190493/189790864-a164f358-e44f-43ec-9117-d797dc997f8f.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by rej55** [comment]
I can't reproduce it :thinking:

**Comment by taikitanaka3** [comment]
@yukkysaito 
can you tell me commit hash? if your branch is old then occlusion spot could have inserted stop line
https://github.com/autowarefoundation/autoware.universe/pull/1541#issuecomment-1240855665

**Comment by taikitanaka3** [comment]
@rej55 
I think this additional change is necessary to launch with awf:autoware.universe
image
image

**Comment by rej55** [comment]
@taikitanaka3 
Thank you. I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/1541/commits/4ad38517b413c5489cff5cd0ba0a39213ec7bff9

**Comment by shmpwk** [comment]
@rej55 
I'm curious about how to send `cooperate_commands`. Do you just publish the topic from your command line?

File: launch/tier4_planning_launch/launch/scenario_planning/lane_driving.launch.xml [hunk 1]
```diff
@@ -25,11 +25,11 @@
       </group>
       <group>
         <arg
-          name="rtc_auto_approver_param_path"
-          default="$(find-pkg-share tier4_planning_launch)/config/scenario_planning/lane_driving/behavior_planning/rtc_auto_approver/rtc_auto_approver.param.yaml"
+          name="rtc_auto_mode_manager_param_path"
+          default="$(find-pkg-share planning_launch)/config/scenario_planning/lane_driving/behavior_planning/rtc_auto_mode_manager/rtc_auto_mode_manager.param.yaml"
```
Comment (taikitanaka3) [inline launch/tier4_planning_launch/launch/scenario_planning/lane_driving.launch.xml:29]:
```suggestion
          default="$(find-pkg-share tier4_planning_launch)/config/scenario_planning/lane_driving/behavior_planning/rtc_auto_mode_manager/rtc_auto_mode_manager.param.yaml"
```

**Comment by yukkysaito** [comment]
>I can't reproduce it 

> can you tell me commit hash? if your branch is old then occlusion spot could have inserted stop line

@rej55 @taikitanaka3 It is the screenshot of movie in PR description.

**Comment by rej55** [comment]
Ok, I see.
This video is the old version of autoware.universe.
I'll update the video in the description.

**Comment by taikitanaka3** [comment]
@yukkysaito 
it's fixed by this PR
https://github.com/autowarefoundation/autoware.universe/pull/1615

**Comment by taikitanaka3** [comment]
@satoshi-ota 
Can you test crosswalk for this PR ?
I have tested with awf: autoware_launch but only yield is generated and there is stop behavior for crosswalk
https://user-images.githubusercontent.com/65527974/189782931-0708cb9d-ce92-4ca4-8805-8c8da56374a5.mp4

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Review by taikitanaka3 — APPROVED** [review]

**Comment by rej55** [comment]
@TakaHoribe 
Could you approve this change?

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**